lib/todolist.ex

```diff
@@ -2,25 +2,25 @@ defmodule TodoList do
   @moduledoc """
   Simple todo-list
   """
-  defimpl(String.Chars, do: def(to_string(this), do: inspect(this)))
 
+  @typedoc "Entry format"
+  @type entry :: %{date: Date.t(), id: non_neg_integer(), title: String.t()}
   @typedoc "The type of todolist"
   @type t :: %{
           __struct__: TodoList,
-          auto_id: number(),
-          entries: map()
+          auto_id: non_neg_integer(),
+          entries: %{optional(non_neg_integer()) => entry()}
         }
 
   defstruct auto_id: 1, entries: %{}
 
-
   @doc """
   Create new Todo list. Empty if without params, or from initial list.
   ## Examples
       iex> TodoList.new()
       %TodoList{auto_id: 1, entries: %{}}
   """
-  @spec new(any) :: Todolist.t() # fix any
+  @spec new(Enumerable.t()) :: t()
```

Comment:
You iterate over entries, hence all you need is `Enumerable.t()` (`Stream`, `list`, `map`, or like.)